crates/solvers/src/infra/dex/oneinch/mod.rs

```diff
@@ -68,6 +68,14 @@ impl OneInch {
                 let status = response.status();
                 let body = response.text().await?;
                 tracing::trace!(status = %status.as_u16(), %body, "fetched 1inch liquidity sources");
+
+                if !status.is_success() {
+                    return Err(Error::Api {
+                        code: status.as_u16() as i32,
```

Comment:
Are these casts actually necessary since a `u16` can be represented as an `i64`?

Reply:
<img width="640" alt="image" src="https://github.com/cowprotocol/services/assets/1200333/633936fc-4237-4107-ade8-172a321aa822">